src/Illuminate/Queue/Jobs/Job.php

```diff
@@ -194,6 +195,24 @@ public function payload()
         return json_decode($this->getRawBody(), true);
     }
 
+    /**
+     * Get the value from the shared data or return the default.
+     *
+     * @param  string|null  $key
+     * @param  mixed  $default
+     * @return \Illuminate\Support\Collection|mixed
+     */
+    public function shared($key, $default = null)
+    {
+        $shared = Collection::make($this->payload()['shared'] ?? []);
```

Comment:
This check is not necessary you can leave it as:
$shared = Collection::make($this->payload()['shared']);